### PR TITLE
ADD: useFetchData hook

### PR DESCRIPTION
**Pull request by JDeepD**
```
This custom hook takes an API endpoint as an argument and returns a stateful response of the following structure:

{
    loading: bool,
    response: Array [],
    error:  Error
}

loading is True until response or error is set.
```

### CODE REVIEW

**Comment by JDeepD** [comment]
Along with the hook, this commit also modifies `TagItem` component to redirect to an url specified in the json response.

File: src/Hooks/useFetchData.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        response: jresp.status === 200 ? jresp.msg : [],
+        error: jresp.status !== 200 ? jresp.msg : null,
+      };
+    });
+  } catch (e) {
+    setState((prevState) => {
+      return {
+        ...prevState,
+        loading: false,
+        error: e,
```
Comment (codadept) [inline src/Hooks/useFetchData.js]: Return `e.message` as `e` is a object of the `Error` class in Javascript.

File: src/Hooks/useFetchData.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      };
+    });
+  }
+};
+
```
Comment (codadept) [inline src/Hooks/useFetchData.js:25]: Write a small documentation on the usage of the function.

**Review by codadept — CHANGES_REQUESTED** [review]
Do these changes, rest good to go!

**Review by codadept — APPROVED** [review]

**Pull request merged**